lib/server.js

```diff
@@ -15,385 +15,366 @@ const userAgent = require('./userAgent')
-    log.error(endpointLog)
-  }
+    var parts = header.split(',')
+    var str = parts[0]
+    if (str.length >= 255) { return null }
+    for (var i = 1; i < parts.length && str.length + parts[i].length < 255; i++) {
+        str += ',' + parts[i]
+    }
+    return str.trim()
 }
 
-function create(log, error, config, routes, db, translator) {
-  const getGeoData = require('./geodb')(log)
-
-  // Hawk needs to calculate request signatures based on public URL,
-  // not the local URL to which it is bound.
-  var publicURL = url.parse(config.publicUrl)
-  var defaultPorts = {
-    'http:': 80,
-    'https:': 443
-  }
-  var hawkOptions = {
-    host: publicURL.hostname,
-    port: publicURL.port ? publicURL.port : defaultPorts[publicURL.protocol],
-
-    // We're seeing massive clock skew in deployed clients, and it's
-    // making auth harder than it needs to be.  This effectively disables
-    // the timestamp checks by setting it to a humongous value.
-    timestampSkewSec: 20 * 365 * 24 * 60 * 60,  // 20 years, +/- a few days
-
-    nonceFunc: function nonceCheck(key, nonce, ts, cb) {
-      // Since we've disabled timestamp checks, there's not much point
-      // keeping a nonce cache.  Instead we use this as an opportunity
-      // to report on the clock skew values seen in the wild.
-      var skew = (Date.now() / 1000) - (+ts)
-      log.trace({ op: 'server.nonceFunc', skew: skew })
-      return cb()
+function logEndpointErrors(response, log) {
+    // When requests to DB timeout and fail for unknown reason they are an 'EndpointError'.
+    // The error response hides error information from the user, but we log it here
+    // to better understand the DB timeouts.
+    if (response.__proto__ && response.__proto__.name === 'EndpointError') {
+        var endpointLog = {
+            op: 'server.EndpointError',
+            message: response.message,
+            reason: response.reason
+        }
+        if (response.attempt && response.attempt.method) {
+            // log the DB attempt to understand the action
+            endpointLog.method = response.attempt.method
+        }
+        log.error(endpointLog)
     }
-  }
-
-  function makeCredentialFn(dbGetFn) {
-    return function (id, cb) {
-      log.trace({ op: 'DB.getToken', id: id })
-      if (! HEX_STRING.test(id)) {
-        return process.nextTick(cb.bind(null, null, null)) // not found
-      }
-      dbGetFn(id)
-        .then(token => {
-          if (token.expired(Date.now())) {
-            const err = error.invalidToken('The authentication token has expired')
-
-            if (token.tokenTypeID === 'sessionToken') {
-              return db.pruneSessionTokens(token.uid, [ token ])
-                .catch(() => {})
-                .then(() => cb(err))
-            }
+}
 
-            return cb(err)
-          }
+async function create (log, error, config, routes, db, translator) {
+    const getGeoData = require('./geodb')(log)
 
-          return cb(null, token)
```

Comment:
@deeptibaghel I think you deleted the "no error" case, so now everything is unauthenticated 🔍